File: data/export_properties.py
import os
import csv
import logging
from pymongo import MongoClient
from dotenv import load_dotenv

# Load environment variables.
load_dotenv()

MONGO_URI = os.getenv("MONGO_URI")
if not MONGO_URI:
    logging.error("MONGO_URI is not set in .env")
    exit(1)

# Connect to MongoDB.
client = MongoClient(MONGO_URI)
db = client.get_default_database()
properties_collection = db["properties"]


def export_properties_to_csv(filename):
    """
    Exports all property documents to a CSV file.
    """
    cursor = properties_collection.find({})
    fieldnames = [
        "zpid",
        "city",
        "state",
        "homeStatus",
        "address",
        "bedrooms",
        "bathrooms",
        "price",
        "yearBuilt",
        "latitude",
        "longitude",
        "livingArea",
        "homeType",
        "listingDataSource",
        "description"
    ]

    with open(filename, "w", newline="", encoding="utf-8") as csvfile:
        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
        writer.writeheader()
        for doc in cursor:
            # Convert 'address' from dict to string if necessary.
            if "address" in doc and isinstance(doc["address"], dict):
                doc["address"] = str(doc["address"])
            writer.writerow({k: doc.get(k, "") for k in fieldnames})

    print(f"Data exported successfully to {filename}")


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO)
    export_properties_to_csv("properties_export.csv")
    client.close()

File: data/js/exportProperties.js
require("dotenv").config();
const mongoose = require("mongoose");
const fs = require("fs");
const { Parser } = require("json2csv");

const propertySchema = new mongoose.Schema({}, { strict: false });
const Property = mongoose.model("Property", propertySchema, "properties");

const MONGO_URI = process.env.MONGO_URI;
if (!MONGO_URI) {
  console.error("MONGO_URI not defined in .env");
  process.exit(1);
}

mongoose
  .connect(MONGO_URI)
  .then(() => {
    console.log("Connected to MongoDB");
    exportToCSV();
  })
  .catch((err) => {
    console.error("Error connecting to MongoDB", err);
    process.exit(1);
  });

async function exportToCSV() {
  try {
    const properties = await Property.find({}).lean();
    if (!properties || properties.length === 0) {
      console.log("No property data found");
      return;
    }

    // Define CSV fields – adjust as needed.
    const fields = [
      "zpid",
      "city",
      "state",
      "homeStatus",
      "address.streetAddress",
      "address.city",
      "address.state",
      "address.zipcode",
      "bedrooms",
      "bathrooms",
      "price",
      "yearBuilt",
      "latitude",
      "longitude",
      "livingArea",
      "homeType",
      "listingDataSource",
      "description",
    ];
    const opts = { fields };

    // Convert JSON to CSV.
    const parser = new Parser(opts);
    const csv = parser.parse(properties);

    // Write CSV data to file.
    fs.writeFileSync("properties_export.csv", csv);
    console.log("Exported properties to properties_export.csv");
  } catch (error) {
    console.error("Error exporting properties", error);
  } finally {
    mongoose.connection.close();
  }
}
